sqlmesh/core/plan/definition.py

```diff
@@ -378,11 +394,11 @@ def set_choice(self, snapshot: Snapshot, choice: SnapshotChangeCategory) -> None
             else:
                 child_snapshot.categorize_as(SnapshotChangeCategory.INDIRECT_NON_BREAKING)
 
-            snapshot.indirect_versions[child] = child_snapshot.all_versions
+            snapshot.indirect_versions[child_s_id.name] = child_snapshot.all_versions
 
             for upstream in self.directly_modified:
-                if child in upstream.indirect_versions:
-                    data_version = upstream.indirect_versions[child][-1]
+                if child_s_id.name in upstream.indirect_versions:
```

Comment:
Did `indirect_versions` keep the model name (not fqn) as a name?

Reply:
Well snapshot name but yes. This is because it is part of Snapshot which is serialized and therefore we can't key off of SnapshotId.